manifests/package/debian.pp

```diff
@@ -32,6 +32,10 @@
     include '::apt'
     Exec['apt_update'] -> Package['nginx']
 
+    ensure_packages([ 'apt-transport-https', 'ca-certificates' ])
```

Comment:
Maybe should only ensure it when `manage_repo` is true, since there's still the use-case where the user is managing repos separately or using the vendor package.

I've been meaning to rework package management in general, as described in #938 (or possibly a replacement for it) but not sure when I'll have the time. I got some helpful input from @hunner but somehow the notes didn't seem to make it into the PR comments.